src/main.rs

```diff
@@ -1,11 +1,88 @@
-mod constants;
+use std::{
+    env,
+    fs::File,
+    io::{self, BufReader},
+    process::exit,
+};
+
+use byteorder::{BigEndian, ReadBytesExt};
+use memory::Memory;
+use registers::Registers;
+use termios::{
+    tcsetattr, Termios, BRKINT, ECHO, ICANON, ICRNL, IGNBRK, IGNCR, INLCR, ISTRIP, IXON, PARMRK,
+    TCSANOW,
+};
+use vm::Vm;
+
 mod instructions;
 mod memory;
 mod opcodes;
 mod registers;
+mod traps;
 mod utils;
 mod vm;
 
+const STDIN: i32 = 0;
+
+fn setup_terminal(termios: Termios) {
+    let mut new_termios = termios;
+    new_termios.c_iflag &= IGNBRK | BRKINT | PARMRK | ISTRIP | INLCR | IGNCR | ICRNL | IXON;
+    new_termios.c_lflag &= !(ICANON | ECHO);
+
+    tcsetattr(STDIN, TCSANOW, &new_termios).unwrap();
+}
+
+fn restore_terminal(termios: Termios) {
+    tcsetattr(STDIN, TCSANOW, &termios).unwrap();
+}
```

Comment:
Maybe these functions could be in `utils.rs` ?

Reply:
You're right, they belong there